modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -268,8 +279,23 @@ public void visit(BLangAction actionNode) {
         BInvokableSymbol actionSymbol = Symbols
                 .createActionSymbol(Flags.asMask(actionNode.flagSet), names.fromIdNode(actionNode.name),
                         env.enclPkg.symbol.pkgID, null, env.scope.owner);
+        // Create and keep parameter node to inject as action first parameter later
+        BLangVariable connectorParam = (BLangVariable) TreeBuilder.createVariableNode();
+        connectorParam.pos = env.node.pos;
+        connectorParam.setName(this.createIdentifier(Names.CONNECTOR.getValue()));
+        BLangUserDefinedType connectorType = (BLangUserDefinedType) TreeBuilder.createUserDefinedTypeNode();
```

Comment:
Do we need to keep the connector variable node here? We just need the connector symbol right?